steward/src/cellars/cellar_v2.rs

```diff
@@ -0,0 +1,702 @@
+
+                            debug!("encoded: {:?}", hex::encode(&swap_params));
+                            let call = steward_abi::aave_debt_token_adaptor::SwapCall {
+                                asset_in: sp_call_parse_address(p.asset_in)?,
+                                asset_out: sp_call_parse_address(p.asset_out)?,
+                                amount_in: string_to_u256(p.amount_in)?,
+                                exchange: convert_exchange(p.exchange),
+                                params: swap_params.into(),
+                            };
+                            calls.push(AaveDebtTokenAdaptorCalls::Swap(call).encode().into())
+                        }
+                        aave_debt_token_adaptor::Function::OracleSwap(p) => {
+                            let oracle_swap_params =
+                                encode_oracle_swap_params(p.params.ok_or_else(|| {
+                                    ErrorKind::SPCallError.context("swap params cannot be empty")
+                                })?)?;
+
+                            debug!("encoded: {:?}", hex::encode(&oracle_swap_params));
+                            let call = steward_abi::aave_debt_token_adaptor::OracleSwapCall {
+                                asset_in: sp_call_parse_address(p.asset_in)?,
+                                asset_out: sp_call_parse_address(p.asset_out)?,
+                                amount_in: string_to_u256(p.amount_in)?,
+                                exchange: convert_exchange(p.exchange),
+                                params: oracle_swap_params.into(),
+                                slippage: p.slippage,
+                            };
+                            calls.push(AaveDebtTokenAdaptorCalls::OracleSwap(call).encode().into())
+                        }
+                        aave_debt_token_adaptor::Function::RevokeApproval(p) => {
+                            let call = steward_abi::aave_debt_token_adaptor::RevokeApprovalCall {
+                                asset: sp_call_parse_address(p.asset)?,
+                                spender: sp_call_parse_address(p.spender)?,
+                            };
+                            calls.push(
+                                AaveDebtTokenAdaptorCalls::RevokeApproval(call)
+                                    .encode()
+                                    .into(),
+                            )
+                        }
+                    }
+                }
+            }
+            CallData::CompoundCTokenCalls(params) => {
+                for c in params.calls {
+                    match c.function.expect("adaptor function was empty") {
+                        compound_c_token_adaptor::Function::DepositToCompound(p) => {
+                            let call =
+                                steward_abi::compound_c_token_adaptor::DepositToCompoundCall {
+                                    market: sp_call_parse_address(p.market)?,
```

Comment:
It seems like based on the generated ABI `CErc20` for Compound are treated the same as `ERC20` in types of expecting an address. None of our mock calls are using that type though -- it would be great if we could confirm this serializes as expected.

Reply:
It seems like the ABI calling them the same type is definitive enough to me, but I can add this tomorrow if you disagree.